editor/property_editor.cpp

```diff
@@ -2952,14 +2953,22 @@ void PropertyEditor::update_tree() {
 			if (!found) {
 				DocData *dd = EditorHelp::get_doc_data();
 				Map<String, DocData::ClassDoc>::Element *E = dd->class_list.find(classname);
-				if (E) {
-					for (int i = 0; i < E->get().properties.size(); i++) {
-						if (E->get().properties[i].name == propname.operator String()) {
-							descr = E->get().properties[i].description.strip_edges().word_wrap(80);
+				while (inheritance) {
```

Comment:
I see two issues:
- We need to break out of the while loop once a description is found, no need to check all the inheritance tree and maybe fetch a lower level but less accurate property.
- The boolean seems redundant, you're just re-implementing an infinite loop with `break;` statements :)

I'd do something like that instead (untested):
```
while (E && descr == String()) {
    for (int i = 0; i < E->get().properties.size(); i++) {
        if (E->get().properties[i].name == propname.operator String()) {
            descr = E->get().properties[i].description.strip_edges().word_wrap(80);
            break;
        }
    }
    if (!E->get().inherits.empty()) {
        E = dd->class_list.find(E->get().inherits);
    } else {
        break;
    }
}
```

Reply:
Well, i think you're totally right! Thanks for pointing it out... i'll test it and commit again.